pulsar-io/kafka-connect-adaptor/src/main/java/org/apache/pulsar/io/kafka/connect/KafkaConnectSink.java

```diff
@@ -255,11 +254,25 @@ protected SinkRecord toSinkRecord(Record<GenericObject> sourceRecord) {
                 && sourceRecord.getSchema().getSchemaInfo() != null
                 && sourceRecord.getSchema().getSchemaInfo().getType() == SchemaType.KEY_VALUE) {
             KeyValueSchema kvSchema = (KeyValueSchema) sourceRecord.getSchema();
-            KeyValue kv = (KeyValue) sourceRecord.getValue().getNativeObject();
             keySchema = PulsarSchemaToKafkaSchema.getKafkaConnectSchema(kvSchema.getKeySchema());
             valueSchema = PulsarSchemaToKafkaSchema.getKafkaConnectSchema(kvSchema.getValueSchema());
-            key = kv.getKey();
-            value = kv.getValue();
+
+            Object nativeObject = sourceRecord.getValue().getNativeObject();
+
+            if (nativeObject instanceof org.apache.pulsar.common.schema.KeyValue) {
+                org.apache.pulsar.common.schema.KeyValue kv = (org.apache.pulsar.common.schema.KeyValue) nativeObject;
+                key = kv.getKey();
+                value = kv.getValue();
```

Comment:
This is non possible.
The wrapper for KeyValue is created in AutoConsumeSchema.

This class is like a utility class in Pulsar IO. Probably there is some mock test that creates the wrong object.

I believe we can safely drop this

Reply:
@eolivelli did you mean that org.apache.pulsar.io.core.KeyValue is not possible?

Reply:
I'd leave this anyway until we decide if it is ok to get rid of io.core.KeyValue altogether as a fix for https://github.com/apache/pulsar/issues/15027

Reply:
@eolivelli fixed